fix(docs-infra): yet more style fixes/improvements for angular.io

gkalpak opened the pull request:
Yet more style fixes/improvements for angular.io:

<details>
  <summary><b>fix(docs-infra): do not underline link icons on hover</b> <i>(click to expand)</i></summary><br />

  The external links in the docs content and the download links in the "Press kit" page have icons next to their text (an external link icon and a download icon respectively).

  Previously, when hovering over one of those links, the icons used to get underlined along with the link text, which was not desirable.

  This commit fixes it by changing the mechanics of how these icons are positioned inside the anchor elements so that only the link text is underlined on hover.

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109340455-b64bf180-7871-11eb-923b-3113a237b8d8.png" alt="external links on hover before" width="500" />
  </p>

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109340461-b815b500-7871-11eb-9c9d-91b6ffb17346.png" alt="presskit links on hover before" width="500" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109340458-b77d1e80-7871-11eb-884c-c1093ec83e66.png" alt="external links on hover after" width="500" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109340462-b815b500-7871-11eb-8f00-702b2b61f6ac.png" alt="presskit links on hover after" width="500" />
  </p>

</details>

---
<details>
  <summary><b>fix(docs-infra): use a fixed top-menu on marketing pages as well</b> <i>(click to expand)</i></summary><br />

  Previously, in contrast to docs pages, marketing pages had a non-fixed top-menu, which meant that the top-menu would scroll out of the viewport along with the rest of the content. This had a couple of downsides:
  - The UI was different between pages (i.e. different top-menu behavior on docs vs marketing pages).
  - Since some of the marketing pages are long, it was not easy for people to navigate to a different page (i.e. they had to scroll all the way back up).

  This commit improves the UX by using the same, fixed top-menu on all pages, which restores consistency and allows the user to navigate around more easily.

  NOTE:
  The old behavior (non-fixed top-menu) is kept on the homepage, since its top-menu design in a little different than other pages (e.g. it uses a transparent top-menu) and would not play well with a fixed top-menu.

</details>

---
<details>
  <summary><b>fix(docs-infra): improve the layout of the "Features" page on medium screens</b> <i>(click to expand)</i></summary><br />

  The "Features" page organizes features in groups/rows of 3 features each. On wide screens, all 3 paragraphs of a group/row can be shown next to each other. On narrow screens, the layout changes to stack the paragraphs vertically. On medium screens, however, the 3rd paragragh is wrapped over to the next line.

  Previously, the wrapped content was left-aligned, which left a lot of empty space on the right.

  This commit improves the layout on medium screens by ensuring the paragraphs are horizontally centered (with space distributed evenly around them).

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109344670-b64ef000-7877-11eb-9013-890562ff2f3d.png" alt="features page before" height="250" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109344678-b7801d00-7877-11eb-9224-d7715f7d7235.png" alt="features page after" height="250" />
  </p>

</details>

---
<details>
  <summary><b>fix(docs-infra): fix styling of `&lt;code&gt;` elements on older browsers</b> <i>(click to expand)</i></summary><br />
  
  Previously, styling of `<code>` elements utilized the `:not()` CSS pseudo-class with multiple selectors (`:not(h1, h2, ...)`). It turns out that older browsers (such as IE11) do not support multiple selectors in a single `:not()` instance.
  (See [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/:not#description:~:text=Using%20two%20selectors) and [CanIUse](https://caniuse.com/css-not-sel-list) for more info.)

  This commit fixes `<code>` styling to use multiple separate `:not()` instances instead (`:not(h1):not(h2)...`), so that they are styled correctly on older browsers as well.

  NOTE:
  This change seems to trigger some kind of bug in LightHouse that causes the a11y score of `/start` to be calculated as 0 (which is clearly wrong). This happens on Linux (tested on CI and locally using the Windows Subsystem for Linux (WSL)) - on Windows the score is computed correctly as 98/100.
  ([Example failure](https://circleci.com/gh/angular/angular/931038))

  The bug seems to be related to the layout of the content and goes away if we change the viewport size (for example, switching to LightHouse's `desktop` config) or make another change that affects the content's layout (for example, reducing the padding of `<code>` elements).

  To work around the issue, this commit updates the `test-aio-a11y.js` script to test `/start-routing` instead of `/start`.

</details>

---
<details>
  <summary><b>fix(docs-infra): fix `&lt;code&gt;` styling in the "Cheat sheet" guide</b> <i>(click to expand)</i></summary><br />
  
  Since #40944, `<code>` elements have a distinctive background to make them stand out from regular text. However, this styling is not desirable in certain cells of the "Cheat sheet" guide's tables, which exclusively contain code.

  This commit updates the `<code>` styling to remove the distinctive background in those "Cheat sheet" cells.

  _**Before #40944:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109383633-4f215200-78f0-11eb-95e2-2a2c4fb3a31f.png" alt="cheatsheet code before PR 40944" height="250" />
  </p>

  _**Since #40944:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109383634-50527f00-78f0-11eb-8c48-ff5f96918c4d.png" alt="cheatsheet code since PR 40944" height="250" />
  </p>

  _**After this commit:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109383635-50eb1580-78f0-11eb-9cf2-98851692ddd9.png" alt="cheatsheet code after" height="250" />
  </p>

</details>

---
<details>
  <summary><b>refactor(docs-infra): remove unnecessary wrapper `&lt;div&gt;` from the "Cheat sheet" guide</b> <i>(click to expand)</i></summary><br />

  This commit removes an unnecessary wrapper `<div>` from the "Cheat sheet" guide. The CSS styles that referenced the element's ID (`#cheatsheet`) have been updated to use `.page-guide-cheatsheet` instead.

</details>

---
<details>
  <summary><b>fix(docs): fix indentation of code snippets in the "Cheat sheet" guide</b> <i>(click to expand)</i></summary><br />

  Previously, the indentation of code snippets in the "Cheat sheet" guide was done using regular spaces. However, leading whitespace is ignored in HTML elements (by default), which resulted in the identation being lost.

  This commit fixes this by using the `&emsp;` HTML entity for indentation, ensuring the code snippets are easier to read.

  Some examples:

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109385242-218dd600-78fb-11eb-9261-18ab5f2c308c.png" alt="cheatsheet code indentation before" height="250" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109385246-25b9f380-78fb-11eb-8e82-d8a561deba21.png" alt="cheatsheet code since PR 40944" height="250" />
  </p>

</details>

---
<details>
  <summary><b>refactor(docs-infra): fix code indentation in `ContributorListComponent`'s template</b> <i>(click to expand)</i></summary><br />

  This commit fixes the indentation of the template of `ContributorListComponent` to make it more readable.

</details>

---
<details>
  <summary><b>refactor(docs-infra): use more consistent content width on marketing pages</b> <i>(click to expand)</i></summary><br />

  Previously, each marketing page used a different limit for its content's width (if it had a limit at all) and implemented the width limiting in a different way. Besides resulting in an inconsistent UX, this also made it difficult to apply site-wide layout changes.

  This commit makes the limit for most marketing pages consistent and uses the same CSS class to make it easier to apply site-wide changes in the future. The chosen limit is slightly larger than that of docs pages (62.5em/1000px vs 50em/800px), because marketing pages have a different type of content and layout (i.e. images, multi-column layout, etc.). Finally, this commit also removes obsolete wrapper elements, CSS classes and CSS styles, that are no longer necessary after the changes.

  Notably, the homepage (`/`) and the "Contributors" page (`/about`) have remained unchanged, because the former has its own layout that is different from other marketing pages and the latter would offer a worse UX with a small content width limit (as the one used on other marketing pages).

  The content widths of the rest of the marketing pages change slightly as a result of the changes in this commit, but not in a way that would have a negative impact on UX. More specifically:

  | Page (URL)    | Size before | Size after |
  |:--------------|------------:|-----------:|
  | `/contribute` |       880px |     1000px |
  | `/events`     |   unlimited |     1000px |
  | `/features`   |       996px |     1000px |
  | `/presskit`   |       800px |     1000px |
  | `/resources`  |       800px |     1000px |

</details>

---
<details>
  <summary><b>fix(docs-infra): limit max content width on "Contributors" and "Cheat sheet" pages</b> <i>(click to expand)</i></summary><br />

  Previously, the content width on the "Contributors" page (`/about`) and the "Cheat sheet" guide (`/guide/cheatsheet`) was unlimited (i.e. the content would take up all the available space horizontally). This was not very ergonomic on large screens. Especially on the "Cheat sheet" guide there was a lot of unnecessary whitespace between the code shown on the left column and the description shown on the right column.

  This commit fixes this by setting a max content width for both pages of 84em (1404px by default). This keeps the content at a more manageable width and avoids unnecessary whitespace.

  _**Before (contributors):**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109394323-1bb1e800-792f-11eb-9cd6-c5b83e5ae921.png" alt="contributors page on 2500px before" height="250" />
  </p>

  _**Before (cheatsheet):**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109394329-1fde0580-792f-11eb-91c0-4ed7ad408b17.png" alt="cheatsheet on 2500px before" height="250" />
  </p>

  _**After (contributors):**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109394319-181e6100-792f-11eb-9463-a281dfe80448.png" alt="contributors page on 2500px after" height="250" />
  </p>

  _**After (cheatsheet):**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109394330-210f3280-792f-11eb-9837-ddc7a8d01d19.png" alt="cheatsheet on 2500px after" height="250" />
  </p>

</details>

---
<details>
  <summary><b>fix(docs-infra): show ellipsis when text overflows in API list items</b> <i>(click to expand)</i></summary><br />

  Previously, when an API list item's text overflowed its container, it was just hidden. This made it often difficult to realize that there was more text.

  This commit fixes this by ensuring an ellipsis (`...`) is shown when the text overflows.

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109396186-0cd03300-7939-11eb-921c-00621a3889a4.png" alt="api-list truncated before" height="250" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109396187-0e016000-7939-11eb-9f3b-4535be083417.png" alt="api-list truncated after" height="250" />
  </p>

</details>

---
<details>
  <summary><b>fix(docs-infra): avoid truncating several API list items</b> <i>(click to expand)</i></summary><br />

  Previously, with the min width of 220px per item, several API list items were truncated.

  This commit increases the min width per item to 330px, which allows almost all items to have their full text shown. It also increases the API list page's max content width from 50em (800px) to 62.5em (1000px) to allow items to be shown on three columns despite their increased width. This increase in the content width shouldn't negatively affect UX, since the API list page uses a multi-column layout (i.e. it does not contain 1000px-lines of text.)

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109396457-5f5e1f00-793a-11eb-80cf-1418f409325a.png" alt="api-list before" height="250" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109396659-499d2980-793b-11eb-95d3-f54250f7fab5.png" alt="api-list after" height="250" />
  </p>

</details>

---
New commits to address review feedback:

<details>
  <summary><b>fix(docs-infra): improve the layout of the "Features" page on medium screens</b> <i>(click to expand)</i></summary><br />

  The "Features" page organizes features in groups/rows of 3 features each. On wide screens, all 3 paragraphs of a group/row can be shown next to each other. On narrow screens (between 768px and 1057px), the layout changes to stack the paragraphs vertically. On medium screens, however, there is not enough space to show more than two paragraphs next to each other.

  Previously, the 3rd paragraph was wrapped over to the next line.

  This commit improves the layout on medium screens by switching to immediately stacking the paragraphs vertically as soon as there is not enough space for them to be displayed in one row. Since the total width is still too much for one paragraph, the paragraphs are limited to 80% of the total width.

  _**Before (on 1000px width):**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109825316-62128a00-7c42-11eb-8391-650201257274.png" alt="features page (on 1000px) before" height="250" />
  </p>

  _**After (on 1000px width):**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/109825323-6343b700-7c42-11eb-86c1-e8307c5a727a.png" alt="features page (on 1000px) after" height="250" />
  </p>

</details>

---
<details>
  <summary><b>refactor(docs-infra): simplify code snippet indentation in the "Cheat sheet" guide</b> <i>(click to expand)</i></summary><br />

  Previously, the indentation of code snippets in the "Cheat sheet" guide was done using `<br>` elements (for line breaks) and `&emsp;` HTML entities (for space). This was laborious and put the onus on the author to remember to use these symbols (instead of regular whitespace characters).
  (Discussed in #41051 (comment).)

  This commit changes the way `<code>` elements are styles inside the "Cheat sheet" guide to allow using regular whitespace characters in code snippets. It also changes all `<br>`/`&emsp;` occurrences to `\n`/`  ` respectively to make code snippets more readable in the source code.

</details>

mary-poppins commented:
You can preview 26b1233 at https://pr41051-26b1233.ngbuilds.io/.

petebacondarwin commented on aio/src/styles/2-modules/_features.scss:
```diff
@@ -22,6 +22,7 @@
   .feature-row {
     display: flex;
     flex-wrap: wrap;
+    justify-content: space-evenly;
```
Personally, I think we should switch straight from 3 in a row to one per row as the width decreases. The current situation where we have two in the first row and one on the second row looks a bit weird.

gkalpak replied:
I am not sure I agree, but I don't feel strognly about it, so there you go: https://github.com/angular/angular/pull/41051/commits/1a821a4196e6c5c029b37dd9b33f6a2215e3e1a7
(I added a 80% width limit on screens between 768px and 1057px, because they felt too wide for a single item.)

petebacondarwin replied:
I guess neither option is perfect. But thanks for being so considerate.

petebacondarwin commented on aio/content/guide/cheatsheet.md:
```diff
@@ -22,7 +22,7 @@
 </tr>
 <tr>
-<td><code>@<b>NgModule</b>({ declarations: ..., imports: ...,<br>     exports: ..., providers: ..., bootstrap: ...})<br>class MyModule {}</code></td>
+<td><code>@<b>NgModule</b>({<br>&emsp;declarations: ..., imports: ..., exports: ...,<br>&emsp;providers: ..., bootstrap: ...<br>})<br>class MyModule {}</code></td>
```
I feel like this is a laborious solution, which also will put the onus on the author to remember to use these `&emsp;` symbols in the future. Why can't we just change the whitespace handling for these cheatsheet code blocks?

gkalpak replied:
Good call :+1:
I added commit https://github.com/angular/angular/pull/41051/commits/8fbd4747c1b79ba0f2b3c3fccd187a114c33dba2.

petebacondarwin commented on aio/src/styles/1-layouts/_layout-global.scss:
```diff
@@ -12,6 +12,11 @@ body,
+  margin: 0 auto;
+  max-width: 84em;
+}
+
```
❤️

petebacondarwin reviewed (approved):
Generally looks good - I had a couple of non-blocking questions/suggestions.

gkalpak commented on aio/src/styles/2-modules/_presskit.scss:
```diff
@@ -50,14 +50,16 @@
+            padding-right: 3rem;
+            position: relative;
 
-            &:after {
-              font-family: 'Material Icons';
-              content: 'cloud_download';
+            &::after {
+              content: "cloud_download";
+              font-family: "Material Icons";
               @include font-size(24);
-              margin-left: 0.5rem;
+              position: absolute;
+              right: 0;
```
Changing `:after` to `::after` and single quotes to double quotes and sorting `content` and `font-family` alphabetically are not strictly necessary, but I wanted these styles to be consistent with the extrnal link icon styles (since they do the same thing).
The rest are necessary.

mary-poppins commented:
You can preview 8fbd474 at https://pr41051-8fbd474.ngbuilds.io/.

jessicajaniuk reviewed (approved):
LGTM 🍪 

Thanks for doing this!

petebacondarwin commented:
There is something funny going on with the Lighthouse score of `/start`.

gkalpak commented:
Yeah, I noticed that. It passes locally for me and that page has hardly changed at all in this PR 🤷‍♂️
I am looking into it...

mary-poppins commented:
You can preview 5c1f1f4 at https://pr41051-5c1f1f4.ngbuilds.io/.

mary-poppins commented:
You can preview 17de905 at https://pr41051-17de905.ngbuilds.io/.

mary-poppins commented:
You can preview aba1c1a at https://pr41051-aba1c1a.ngbuilds.io/.

gkalpak commented:
@petebacondarwin, I looked into the failure and it seems to be a weird LightHouse bug (on Linux) 🤷‍♂️ 
I added https://github.com/angular/angular/pull/41051/commits/09245f07f2de2cff65ebfe764a1101a27cdbf5f7 to work around it (I don't think it is worth addressing it in a different way.) I also updated https://github.com/angular/angular/pull/41051/commits/2634d35558827881789fb43348d3326eef683a7e with a mention/explanation of this change.

petebacondarwin reviewed (approved):
Reviewed-for: global-docs-approvers

gkalpak committed:
fix(docs-infra): do not underline link icons on hover

The external links in the docs content and the download links in the
"Press kit" page have icons next to their text (an external link icon
and a download icon respectively).

Previously, when hovering over one of those links, the icons used to get
underlined along with the link text, which was not desirable.

This commit fixes it by changing the mechanics of how these icons are
positioned inside the anchor elements so that only the link text is
underlined on hover.

Before: ![external links on hover before][1]
Before: ![presskit links on hover before][2]
After: ![external links on hover after][3]
After: ![presskit links on hover after][4]

[1]: https://user-images.githubusercontent.com/8604205/109340455-b64bf180-7871-11eb-923b-3113a237b8d8.png
[2]: https://user-images.githubusercontent.com/8604205/109340461-b815b500-7871-11eb-9c9d-91b6ffb17346.png
[3]: https://user-images.githubusercontent.com/8604205/109340458-b77d1e80-7871-11eb-884c-c1093ec83e66.png
[4]: https://user-images.githubusercontent.com/8604205/109340462-b815b500-7871-11eb-8f00-702b2b61f6ac.png

gkalpak committed:
fix(docs-infra): improve the layout of the "Features" page on medium screens

The "Features" page organizes features in groups/rows of 3 features
each. On wide screens, all 3 paragraphs of a group/row can be shown next
to each other. On narrow screens, the layout changes to stack the
paragraphs vertically. On medium screens, however, the 3rd paragragh is
wrapped over to the next line.

Previously, the wrapped content was left-aligned, which left a lot of
empty space on the right.

This commit improves the layout on medium screens by ensuring the
paragraphs are horizontally centered (with space distributed evenly
around them).

Before: ![features page before][1]
After: ![features page after][2]

[1]: https://user-images.githubusercontent.com/8604205/109344670-b64ef000-7877-11eb-9013-890562ff2f3d.png
[2]: https://user-images.githubusercontent.com/8604205/109344678-b7801d00-7877-11eb-9224-d7715f7d7235.png

gkalpak committed:
fix(docs-infra): use a fixed top-menu on marketing pages as well

Previously, in contrast to docs pages, marketing pages had a non-fixed
top-menu, which meant that the top-menu would scroll out of the viewport
along with the rest of the content. This had a couple of downsides:
- The UI was different between pages (i.e. different top-menu behavior
  on docs vs marketing pages).
- Since some of the marketing pages are long, it was not easy for people
  to navigate to a different page (i.e. they had to scroll all the way
  back up).

This commit improves the UX by using the same, fixed top-menu on all
pages, which restores consistency and allows the user to navigate around
more easily.

NOTE:
The old behavior (non-fixed top-menu) is kept on the homepage, since its
top-menu design in a little different than other pages (e.g. it uses a
transparent top-menu) and would not play well with a fixed top-menu.

gkalpak committed:
fix(docs-infra): fix styling of `<code>` elements on older browsers

Previously, styling of `<code>` elements utilized the `:not()` CSS
pseudo-class with multiple selectors (`:not(h1, h2, ...)`). It turns out
that older browsers (such as IE11) do not support multiple selectors in
a single `:not()` instance.
(See [MDN][1] and [CanIUse][2] for more info.)

This commit fixes `<code>` styling to use multiple separate `:not()`
instances instead (`:not(h1):not(h2)...`), so that they are styled
correctly on older browsers as well.

NOTE:
This change seems to trigger some kind of bug in LightHouse that causes
the a11y score of `/start` to be calculated as 0 (which is clearly
wrong). This happens on Linux (tested on CI and locally using the
Windows Subsystem for Linux (WSL)) - on Windows the score is computed
correctly as 98/100.
([Example failure][3])

The bug seems to be related to the layout of the content and goes away
if we change the viewport size (for example, switching to LightHouse's
`desktop` config) or make another change that affects the content's
layout (for example, reducing the padding of `<code>` elements).

To work around the issue, this commit updates the `test-aio-a11y.js`
script to test `/start-routing` instead of `/start`.

[1]: https://developer.mozilla.org/en-US/docs/Web/CSS/:not#description:~:text=Using%20two%20selectors
[2]: https://caniuse.com/css-not-sel-list
[3]: https://circleci.com/gh/angular/angular/931038

gkalpak committed:
fixup! fix(docs-infra): fix styling of `<code>` elements on older browsers

gkalpak committed:
refactor(docs-infra): remove unnecessary wrapper `<div>` from the "Cheat sheet" guide

This commit removes an unnecessary wrapper `<div>` from the
"Cheat sheet" guide. The CSS styles that referenced the element's ID
(`#cheatsheet`) have been updated to use `.page-guide-cheatsheet`
instead.

gkalpak committed:
fix(docs-infra): fix `<code>` styling in the "Cheat sheet" guide

Since #40944, `<code>` elements have a distinctive background to make
them stand out from regular text. However, this styling is not desirable
in certain cells of the "Cheat sheet" guide's tables, which exclusively
contain code.

This commit updates the `<code>` styling to remove the distinctive
background in those "Cheat sheet" cells.

Before #40944: ![cheatsheet code before PR 40944][1]
Since #40944: ![cheatsheet code since PR 40944][2]
After this commit: ![cheatsheet code after][3]

[1]: https://user-images.githubusercontent.com/8604205/109383633-4f215200-78f0-11eb-95e2-2a2c4fb3a31f.png
[2]: https://user-images.githubusercontent.com/8604205/109383634-50527f00-78f0-11eb-8c48-ff5f96918c4d.png
[3]: https://user-images.githubusercontent.com/8604205/109383635-50eb1580-78f0-11eb-9cf2-98851692ddd9.png

gkalpak committed:
docs: fix indentation of code snippets in the "Cheat sheet" guide

Previously, the indentation of code snippets in the "Cheat sheet" guide
was done using regular spaces. However, leading whitespace is ignored in
HTML elements (by default), which resulted in the identation being lost.

This commit fixes this by using the `&emsp;` HTML entity for
indentation, ensuring the code snippets are easier to read.

Some examples:

Before: ![cheatsheet code indentation before][1]
After: ![cheatsheet code since PR 40944][2]

[1]: https://user-images.githubusercontent.com/8604205/109385242-218dd600-78fb-11eb-9261-18ab5f2c308c.png
[2]: https://user-images.githubusercontent.com/8604205/109385246-25b9f380-78fb-11eb-8e82-d8a561deba21.png

gkalpak committed:
fix(docs-infra): limit max content width on "Contributors" and "Cheat sheet" pages

Previously, the content width on the "Contributors" page (`/about`) and
the "Cheat sheet" guide (`/guide/cheatsheet`) was unlimited (i.e. the
content would take up all the available space horizontally). This was
not very ergonomic on large screens. Especially on the "Cheat sheet"
guide there was a lot of unnecessary whitespace between the code shown
on the left column and the description shown on the right column.

This commit fixes this by setting a max content width for both pages of
84em (1404px by default). This keeps the content at a more manageable
width and avoids unnecessary whitespace.

Before (contributors): ![contributors page on 2500px before][1]
Before (cheatsheet): ![cheatsheet on 2500px before][2]
After (contributors): ![contributors page on 2500px after][3]
After (cheatsheet): ![cheatsheet on 2500px after][4]

[1]: https://user-images.githubusercontent.com/8604205/109394323-1bb1e800-792f-11eb-9cd6-c5b83e5ae921.png
[2]: https://user-images.githubusercontent.com/8604205/109394329-1fde0580-792f-11eb-91c0-4ed7ad408b17.png
[3]: https://user-images.githubusercontent.com/8604205/109394319-181e6100-792f-11eb-9463-a281dfe80448.png
[4]: https://user-images.githubusercontent.com/8604205/109394330-210f3280-792f-11eb-9837-ddc7a8d01d19.png

gkalpak committed:
refactor(docs-infra): fix code indentation in `ContributorListComponent`'s template

This commit fixes the indentation of the template of
`ContributorListComponent` to make it more readable.

gkalpak committed:
refactor(docs-infra): use more consistent content width on marketing pages

Previously, each marketing page used a different limit for its content's
width (if it had a limit at all) and implemented the width limiting in a
different way. Besides resulting in an inconsistent UX, this also made
it difficult to apply site-wide layout changes.

This commit makes the limit for most marketing pages consistent and uses
the same CSS class to make it easier to apply site-wide changes in the
future. The chosen limit is slightly larger than that of docs pages
(62.5em/1000px vs 50em/800px), because marketing pages have a different
type of content and layout (i.e. images, multi-column layout, etc.).
Finally, this commit also removes obsolete wrapper elements, CSS classes
and CSS styles, that are no longer necessary after the changes.

Notably, the homepage (`/`) and the "Contributors" page (`/about`) have
remained unchanged, because the former has its own layout that is
different from other marketing pages and the latter would offer a worse
UX with a small content width limit (as the one used on other marketing
pages).

The content widths of the rest of the marketing pages change slightly as
a result of the changes in this commit, but not in a way that would have
a negative impact on UX. More specifically:

| Page (URL)    | Size before | Size after |
|:--------------|------------:|-----------:|
| `/contribute` |       880px |     1000px |
| `/events`     |   unlimited |     1000px |
| `/features`   |       996px |     1000px |
| `/presskit`   |       800px |     1000px |
| `/resources`  |       800px |     1000px |

gkalpak committed:
fix(docs-infra): avoid truncating several API list items

Previously, with the min width of 220px per item, several API list items
were truncated.

This commit increases the min width per item to 330px, which allows
almost all items to have their full text shown. It also increases the
API list page's max content width from 50em (800px) to 62.5em (1000px)
to allow items to be shown on three columns despite their increased
width. This increase in the content width shouldn't negatively affect
UX, since the API list page uses a multi-column layout (i.e. it does not
contain 1000px-lines of text.)

Before: ![api-list before][1]
After: ![api-list after][2]

[1]: https://user-images.githubusercontent.com/8604205/109396457-5f5e1f00-793a-11eb-80cf-1418f409325a.png
[2]: https://user-images.githubusercontent.com/8604205/109396659-499d2980-793b-11eb-95d3-f54250f7fab5.png

gkalpak committed:
fix(docs-infra): show ellipsis when text overflows in API list items

Previously, when an API list item's text overflowed its container, it
was just hidden. This made it often difficult to realize that there was
more text.

This commit fixes this by ensuring an ellipsis (`...`) is shown when the
text overflows.

Before: ![api-list truncated before][1]
After: ![api-list truncated after][2]

[1]: https://user-images.githubusercontent.com/8604205/109396186-0cd03300-7939-11eb-921c-00621a3889a4.png
[2]: https://user-images.githubusercontent.com/8604205/109396187-0e016000-7939-11eb-9f3b-4535be083417.png

gkalpak committed:
fix(docs-infra): improve the layout of the "Features" page on medium screens

The "Features" page organizes features in groups/rows of 3 features
each. On wide screens, all 3 paragraphs of a group/row can be shown next
to each other. On narrow screens (between 768px and 1057px), the layout
changes to stack the paragraphs vertically. On medium screens, however,
there is not enough space to show more than two paragraphs next to each
other.

Previously, the 3rd paragraph was wrapped over to the next line.

This commit improves the layout on medium screens by switching to
immediately stacking the paragraphs vertically as soon as there is not
enough space for them to be displayed in one row. Since the total width
is still too much for one paragraph, the paragraphs are limited to 80%
of the total width.

Before (on 1000px width): [features page (on 1000px) before][1]
After (on 1000px width): [features page (on 1000px) after][2]

[1]: https://user-images.githubusercontent.com/8604205/109825316-62128a00-7c42-11eb-8391-650201257274.png
[2]: https://user-images.githubusercontent.com/8604205/109825323-6343b700-7c42-11eb-86c1-e8307c5a727a.png

gkalpak committed:
refactor(docs-infra): simplify code snippet indentation in the "Cheat sheet" guide

Previously, the indentation of code snippets in the "Cheat sheet" guide
was done using `<br>` elements (for line breaks) and `&emsp;` HTML
entities (for space). This was laborious and put the onus on the author
to remember to use these symbols (instead of regular whitespace
characters).
(Discussed in
https://github.com/angular/angular/pull/41051#discussion_r585651621.)

This commit changes the way `<code>` elements are styles inside the
"Cheat sheet" guide to allow using regular whitespace characters in code
snippets. It also changes all `<br>`/`&emsp;` occurrences to `\n`/`  `
respectively to make code snippets more readable in the source code.

mary-poppins commented:
You can preview d6ad8fb at https://pr41051-d6ad8fb.ngbuilds.io/.
